src/test/datascience/data-viewing/dataWrangler.unit.test.ts

```diff
@@ -0,0 +1,92 @@
+import { ThemeFinder } from '../../../client/datascience/themeFinder';
+import { ICodeCssGenerator, IThemeFinder } from '../../../client/datascience/types';
+import { MockMemento } from '../../mocks/mementos';
+
+suite('DataScience - DataWrangler', () => {
+    let dataWrangler: IDataWrangler;
+    let webPanelProvider: IWebviewPanelProvider;
+    let configService: IConfigurationService;
+    let codeCssGenerator: ICodeCssGenerator;
+    let themeFinder: IThemeFinder;
+    let workspaceService: IWorkspaceService;
+    let applicationShell: IApplicationShell;
+    let dataProvider: IDataViewerDataProvider;
+    let commandManager: CommandManager;
+    let jupyterVariables: KernelVariables;
+    let dataProviderFactory: JupyterVariableDataProviderFactory;
+
+    const title: string = 'Data Wrangler - Title';
+
+    setup(async () => {
+        webPanelProvider = mock(WebviewPanelProvider);
+        configService = mock(ConfigurationService);
+        codeCssGenerator = mock(CodeCssGenerator);
+        themeFinder = mock(ThemeFinder);
+        workspaceService = mock(WorkspaceService);
+        applicationShell = mock(ApplicationShell);
+        dataProvider = mock(JupyterVariableDataProvider);
+        commandManager = mock(CommandManager);
+        jupyterVariables = mock(KernelVariables);
+        dataProviderFactory = mock(JupyterVariableDataProviderFactory);
+        const documentManager = mock(DocumentManager);
+        const settings = mock(JupyterSettings);
+        const settingsChangedEvent = new EventEmitter<void>();
+
+        when(settings.onDidChange).thenReturn(settingsChangedEvent.event);
+        when(configService.getSettings(anything())).thenReturn(instance(settings));
+
+        const configChangeEvent = new EventEmitter<ConfigurationChangeEvent>();
+        when(workspaceService.onDidChangeConfiguration).thenReturn(configChangeEvent.event);
+
+        when(dataProvider.getDataFrameInfo(anything(), anything())).thenResolve({});
+
+        dataWrangler = new DataWrangler(
+            instance(configService),
+            instance(webPanelProvider),
+            instance(codeCssGenerator),
+            instance(themeFinder),
+            instance(workspaceService),
+            instance(applicationShell),
+            false,
+            new MockMemento(),
+            instance(commandManager),
+            instance(documentManager),
+            instance(jupyterVariables),
+            instance(dataProviderFactory)
+        );
+    });
+    test('Data Wrangler showData calls gets dataFrame info from data provider', async () => {
+        await dataWrangler.showData(instance(dataProvider), title);
+
+        verify(dataProvider.getDataFrameInfo(anything(), anything())).once();
+    });
+    test('Data Wrangler calls data provider dispose', async () => {
+        await dataWrangler.showData(instance(dataProvider), title);
+        dataWrangler.dispose();
+
+        verify(dataProvider.dispose()).once();
+    });
```

Comment:
I think we should have an integration test for it? Maybe I missed that.

Reply:
added a simple one